File: moje_rjesenjeV2/webapi/FIT_Api_Examples/Endpoints/StudentEndpoints/StudentMaticnaGetEndpoint.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading;
using System.Threading.Tasks;
using FIT_Api_Example.Helper;
using FIT_Api_Examples.Data;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace FIT_Api_Examples.Endpoints.StudentEndpoints
{
    public class StudentMaticnaGetEndpoint : MyBaseEndpoint<int, StudentMaticnaGetResponse>
    {
        private readonly ApplicationDbContext _applicationDbContext;

        public StudentMaticnaGetEndpoint(ApplicationDbContext applicationDbContext)
        {
            _applicationDbContext = applicationDbContext;
        }

        [HttpGet("StudentMaticna/Get")]
        public override async Task<StudentMaticnaGetResponse> Obradi([FromQuery] int request, CancellationToken cancellationToken)
        {
            var Student = await _applicationDbContext.Student.FindAsync(request);
            if (Student == null)
                throw new Exception("Greska StudentMaticnaGetEndpoint student not found");

            var rez = new StudentMaticnaGetResponse
            {
                StudentId = Student.id,
                StudentIme = Student.ime,
                StudentPrezime = Student.prezime,
                UpisaneGodne = await _applicationDbContext.UpisAkGodina
                .Where(u => u.StudentId == Student.id)
                .Select(u => new UpisaneGodine
                {
                    Id = u.Id,
                    AkademskaGodina = u.AkademskaGodina.opis,
                    GodinaStudija = u.GodinaStudija,
                    Obnova = u.Obnova,
                    ZimskiSemesterUpis = u.ZimskiSemesterUpis,
                    ZimskiSemesterOvjera = u.ZimskiSemesterOvjera,
                    EvidentiraoKorsinik = new EvidentiraoKorsinik
                    {
                        Id = u.KorisnickiNalog.id,
                        KorisnickoIme = u.KorisnickiNalog.korisnickoIme
                    }
                })
                .ToListAsync(cancellationToken)
            };
            return rez;
        }
    }
}

File: moje_rjesenjeV2/angular_app/src/app/Endpoints/student-endpoints/student-maticna-get-endpoint.ts
import { Observable } from 'rxjs';
import { MyBaseEndpoint } from '../my-base-endpoint';
import { Injectable } from '@angular/core';
import { MojConfig } from 'src/app/moj-config';
import { HttpClient } from '@angular/common/http';

@Injectable({ providedIn: 'root' })
export class StudentMaticnaGetEndpoint
  implements MyBaseEndpoint<number, StudentMaticnaGetResponse>
{
  constructor(private httpKlijent: HttpClient) {}
  obradi(request: number): Observable<StudentMaticnaGetResponse> {
    let url =
      MojConfig.adresa_servera + '/StudentMaticna/Get?request=' + request;
    return this.httpKlijent.get<StudentMaticnaGetResponse>(url);
  }
}
export interface StudentMaticnaGetResponse {
  studentId: number;
  studentIme: string;
  studentPrezime: string;
  upisaneGodne: UpisaneGodne[];
}

export interface UpisaneGodne {
  id: number;
  akademskaGodina: string;
  godinaStudija: number;
  obnova: boolean;
  zimskiSemesterOvjera: number;
  zimskiSemesterUpis: string;
  evidentiraoKorsinik: EvidentiraoKorsinik;
}

export interface EvidentiraoKorsinik {
  id: number;
  korisnickoIme: string;
}
